docs/components/DocumentationSidebarRight.js

```diff
@@ -0,0 +1,116 @@
+import * as React from 'react';
+import { css } from 'react-emotion';
+
+import { BASE_HEADING_LEVEL } from '../common/headingManager';
+import DocumentationSidebarRightLink from './DocumentationSidebarRightLink';
+
+import * as Constants from '~/common/constants';
+import { paragraph } from '~/components/base/typography';
+import withHeadingManager from '~/components/page-higher-order/withHeadingManager';
+
+const STYLES_SIDEBAR = css`
+  padding: 20px 24px 24px 24px;
+  width: 280px;
+
+  @media screen and (max-width: ${Constants.breakpoints.mobile}) {
+    width: 100%;
+  }
+`;
+
+const STYLES_TITLE = css`
+  ${paragraph}
+  font-size: 15px;
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  position: relative;
+  margin-bottom: 16px;
+  text-decoration: none;
+  font-family: ${Constants.fontFamilies.demi};
+  user-select: none;
+  background: ${Constants.expoColors.gray[200]};
+  padding: 8px 16px;
+  border-radius: 4px;
+  color: ${Constants.expoColors.black};
+
+  :hover {
+    cursor: pointer;
+  }
+`;
+
+const STYLES_SIDEBAR_INDENT = css`
+  padding-left: 16px;
+`;
+
+class DocumentationSidebarRight extends React.Component {
+  static defaultProps = {
+    maxNestingDepth: 4,
+  };
+
+  state = {
+    activeSlug: null,
+  };
+
+  handleContentScroll(contentScrollPosition) {
+    const { headingManager } = this.props;
+    const { headings } = headingManager;
+
+    for (const heading of headings) {
+      if (!heading.ref) {
+        continue;
+      }
+      const { current } = heading.ref;
+      if (current && current.offsetTop >= contentScrollPosition) {
+        if (heading.slug !== this.state.activeSlug) {
+          this.setState({ activeSlug: heading.slug });
+        }
+        break;
+      }
+    }
+  }
```

Comment:
```suggestion
    for (const { ref, slug } of headings) {
      if (!ref || !ref.current) {
        continue;
      }
      if (ref.current.offsetTop >= contentScrollPosition) {
        this.setState({ activeSlug: slug });
        return;
      }
    }
  }
```
React is not performing re-rendering if the state is not changed. And passing the same (sub-)state via `this.setState` is no-op in this case.